plasmapy/classes/sources/tests/test_openpmd_hdf5.py

```diff
@@ -0,0 +1,46 @@
+import pytest
+import plasmapy
+from astropy import units as u
+import os
+
+@pytest.fixture(scope="module")
+def openPMD2DPlasma():
+    """This is me not knowing how to handle the new Plasma class with data!
+
+    Downloaded from https://github.com/openPMD/openPMD-example-datasets/blob/draft/example-2d.tar.gz
+    per the Creative Commons Zero v1.0 Universal license"""
+    return plasmapy.classes.Plasma(
+        hdf5=os.path.join(
+            os.path.dirname(__file__) or '.',
+            "data",
+            "data00000255.h5")
+        )
+
+def test_has_electric_field_with_units(openPMD2DPlasma):
+    "It would be nice to let the fields have the correct units from the get go"
+    assert openPMD2DPlasma.electric_field.to(u.V / u.m)
+
+def test_has_charge_density_with_units(openPMD2DPlasma):
+    assert openPMD2DPlasma.charge_density.to(u.C/u.m**3)  # unless it's some
+    # 2D charge density
+
+def test_correct_shape_electric_field(openPMD2DPlasma):
+    assert openPMD2DPlasma.electric_field.shape == (3, 51, 201)
+    # IIRC this is how we defined it in the old Plasma class but I can be
+    # convinced otherwise if another way makes more sense
```

Comment:
How about something like this?
```python
assert openPMD2DPlasma.electric_field.x.shape == (51, 201)
assert openPMD2DPlasma.electric_field.y.shape == (51, 201)
assert openPMD2DPlasma.electric_field.z.shape == (51, 201)
```

Reply:
I'm fine with either of them though.

Reply:
That might make sense, but it also might make using the electric field harder in simulations if you can't just access it as one big 3D array... but there may be ways around that. I'm not sure what the optimal choice is here. I think this is a good point to talk about at tomorrow's telecon, if we're all up for that.